model/riscv_insts_zicond.sail

```diff
@@ -0,0 +1,30 @@
+union clause ast = ZICOND_RTYPE : (regidx, regidx, regidx, zicondop)
+
+mapping clause encdec = ZICOND_RTYPE(rs2, rs1, rd, RISCV_CZERO_EQZ) if haveZiCond() <-> 0b0000111 @ rs2 @ rs1 @ 0b101 @ rd @ 0b0110011
+mapping clause encdec = ZICOND_RTYPE(rs2, rs1, rd, RISCV_CZERO_NEZ) if haveZiCond() <-> 0b0000111 @ rs2 @ rs1 @ 0b111 @ rd @ 0b0110011
+
+mapping zicond_mnemonic : zicondop <-> string = {
+  RISCV_CZERO_EQZ <-> "czero.eqz",
+  RISCV_CZERO_NEZ <-> "czero.nez"
+}
+
+mapping clause assembly = ZICOND_RTYPE(rs2, rs1, rd, op)
+  <-> zicond_mnemonic(op) ^ spc() ^ reg_name(rd) ^ sep() ^ reg_name(rs1) ^ sep() ^ reg_name(rs2)
+
+function clause execute (ZICOND_RTYPE(rs2, rs1, rd, RISCV_CZERO_EQZ)) = {
+  let value = X(rs1);
+  let condition = X(rs2);
+  result : xlenbits = if (condition == zeros()) then zeros()
+                                                else value;
+  X(rd) = result;
+  RETIRE_SUCCESS
+}
+
+function clause execute (ZICOND_RTYPE(rs2, rs1, rd, RISCV_CZERO_NEZ)) = {
+  let value = X(rs1);
+  let condition = X(rs2);
+  result : xlenbits = if (condition != zeros()) then zeros()
+                                                else value;
+  X(rd) = result;
+  RETIRE_SUCCESS
+}
```

Comment:
I know the current style matches other extensions, but wouldn't it be simpler to do something like the following (especially since there are only two instructions and the enum doesn't really allow you to share much code):

```suggestion
union clause ast = RISCV_CZERO_EQZ : (regidx, regidx, regidx)
union clause ast = RISCV_CZERO_NEZ : (regidx, regidx, regidx)

mapping clause encdec = RISCV_CZERO_EQZ(rs2, rs1, rd) if haveZiCond() <-> 0b0000111 @ rs2 @ rs1 @ 0b101 @ rd @ 0b0110011
mapping clause encdec = RISCV_CZERO_NEZ(rs2, rs1, rd) if haveZiCond() <-> 0b0000111 @ rs2 @ rs1 @ 0b111 @ rd @ 0b0110011

mapping clause assembly = RISCV_CZERO_EQZ(rs2, rs1, rd)
  <-> "czero.eqz" ^ spc() ^ reg_name(rd) ^ sep() ^ reg_name(rs1) ^ sep() ^ reg_name(rs2)
mapping clause assembly = RISCV_CZERO_NEZ(rs2, rs1, rd)
  <-> "czero.nez" ^ spc() ^ reg_name(rd) ^ sep() ^ reg_name(rs1) ^ sep() ^ reg_name(rs2)

function clause execute (RISCV_CZERO_EQZ(rs2, rs1, rd)) = {
  let value = X(rs1);
  let condition = X(rs2);
  result : xlenbits = if (condition == zeros()) then zeros()
                                                else value;
  X(rd) = result;
  RETIRE_SUCCESS
}

function clause execute (RISCV_CZERO_NEZ(rs2, rs1, rd)) = {
  let value = X(rs1);
  let condition = X(rs2);
  result : xlenbits = if (condition != zeros()) then zeros()
                                                else value;
  X(rd) = result;
  RETIRE_SUCCESS
}
```

Reply:
Let's stick with the same style with have across other extensions for the time being.
My gut tells me that we may eventually want to share the implementation by testing the 'op' inside the function (to determine what comparison operator to use).

That said: the suggested style is indeed much nicer.